datafusion/src/physical_plan/sorts/sort.rs

```diff
@@ -301,17 +306,16 @@ async fn spill_partial_sorted_stream(
 }
 
 fn read_spill_as_stream(
-    path: String,
+    path: NamedTempFile,
```

Comment:
Since ownership of `NamedTempFile` is passed into the actual task doing the reading, so when it is done, the temp file is cleaned up 🧹